components/micro-gateway-core/src/main/ballerina/src/gateway/services/throttle_event_listener.bal

```diff
@@ -70,25 +67,42 @@ service messageServ = service {
 # + return - jms:TopicSubscriber for global throttling event publishing
 public function startSubscriberService() returns @tainted jms:MessageConsumer | error {
     // Initialize a JMS connection  with the provider.
-    jms:Connection connection = check jms:createConnection({
+    jms:Connection | error connection = trap jms:createConnection({
         initialContextFactory: jmsConnectionInitialContextFactory,
         providerUrl: jmsConnectionProviderUrl,
         username: jmsConnectionUsername,
         password: jmsConnectionPassword
 
     });
-    jms:Session session = check connection->createSession({acknowledgementMode: "AUTO_ACKNOWLEDGE"});
-    jms:Destination dest = check session->createTopic("throttleData");
-    jms:MessageConsumer subscriberEndpoint = check session->createDurableSubscriber(dest, "sub-1");
-    var attachResult = subscriberEndpoint.__attach(messageServ);
-    if (attachResult is error) {
-        printDebug(KEY_THROTTLE_UTIL, "Message consumer hasn't been attached to the service.");
-    }
-    var startResult = subscriberEndpoint.__start();
-    if (startResult is error) {
-        printDebug(KEY_THROTTLE_UTIL, "Starting the task is failed.");
+    if (connection is error) {
+        printError(KEY_THROTTLE_UTIL, "Error while creating the jms connection.");
+        return connection;
+    } else {
+        jms:Session | error session = trap connection->createSession({acknowledgementMode: "AUTO_ACKNOWLEDGE"});
+        if (session is error) {
+            printError(KEY_THROTTLE_UTIL, "Error while creating the jms session.");
+            return session;
+        } else {
+        jms:Destination dest = check session->createTopic("throttleData");
+        jms:MessageConsumer | error subscriberEndpoint = trap session->createDurableSubscriber(dest, "sub-1");
+        if (subscriberEndpoint is error) {
+            printError(KEY_THROTTLE_UTIL, "Error while creating the jms subscriber.");
+        } else {
+            var attachResult = subscriberEndpoint.__attach(messageServ);
+            if (attachResult is error) {
+                printDebug(KEY_THROTTLE_UTIL, "Message consumer hasn't been attached to the service.");
```

Comment:
Is it correct to log this as debug level rather than the error

Reply:
added via https://github.com/wso2/product-microgateway/pull/939/commits/ca383ba5c1f48a4101bbf135d8f77449fb77c7a2